### ISSUE TITLE
[TODO] Add some API documentation

### ISSUE THREAD

**Issue by gbowne1** [gbowne1/codebooker#164]
MWe need some API documentation on how to use the API 

Use the /docs/ folder to add API documentation in Markdown

Also they should be added to a section of our Wiki by cloning the Wiki individually and use Markdown to create a section on API.

Add links to documentation in the README.md file

**Comment by amritatwal** [comment]
Hi Gregory, 

I'm interested in helping out with this. I have a particular interest in technical writing. I'm looking for opportunities to practice writing documentation. 

Would you be happy for me to contribute?

**Comment by gbowne1** [comment]
@amritatwal
Yes, this issue is open, so you are welcome to work on this issue or any of the other open documentation issues, and there are several.

I can assign you if you wish to work on this.

**Comment by EdwinChukwuebuka** [comment]
Hello Gregory and amritatwal, I'm also interested in helping out with this as I'm also looking for opportunities to practice api documentation.
Can I contribute too?

**Comment by janetmmiller** [comment]
Hi [@gbowne1](https://github.com/gbowne1), [@amritatwal](https://github.com/amritatwal), [@EdwinChukwuebuka](https://github.com/EdwinChukwuebuka),

Has anyone started on the documentation yet? Maybe we could divide the work into assignments? Message me here. Thanks!

**Comment by EdwinChukwuebuka** [comment]
@janetmmiller not really, just been trying to run the code, understand how it works and test the api on postman.
I should be done with that soon then I'll start the documentation.

**Comment by gbowne1** [comment]
@janetmmiller There are several people wanting to do documentation.

I do not mind if a group of people get together to talk about and/or do any of the documentation for the API or the app itself.  

I would prefer the documentation go on our Wiki but can exist inside the /docs folder 

@LOGESH-B did a great deal of the backend so if you need help with database, APIs, backend etc help he can help you.

I have mainly been doing the Project Management role in this app and helping debug.. due to time limits I have.

**Comment by Chizobaonorh** [comment]
Hello, i would also love to contribute to this documentation. @gbowne1

**Comment by amritatwal** [comment]
Thanks Gregory.

I'd love to still help but I can no longer commit enough time to do this. I'm going to un-assign myself from the issue.

**Comment by gbowne1** [comment]
ok @Chizobaonorh

**Comment by Chizobaonorh** [comment]
> @janetmmiller There are several people wanting to do documentation.
> 
> I do not mind if a group of people get together to talk about and/or do any of the documentation for the API or the app itself.
> 
> I would prefer the documentation go on our Wiki but can exist inside the /docs folder
> 
> @LOGESH-B did a great deal of the backend so if you need help with database, APIs, backend etc help he can help you.
> 
> I have mainly been doing the Project Management role in this app and helping debug.. due to time limits I have.

so we need navigations like where are the endpoints, and what documentation does this project need. How would you see a brief debriefing to get up up to speed on what the documentation requires. @gbowne1

**Comment by gbowne1** [comment]
@Chizobaonorh, @EdwinChukwuebuka etc.

Several things..  

There are some documentation in [Docs](https://github.com/gbowne1/codebooker/tree/master/docs).  

I would like some of the same documentation to appear in the Wiki which can be used for API Documentation, User Documentation and Developer Documentation. [Wiki](https://github.com/gbowne1/codebooker/wiki)

I did not write the API for this application @LOGESH-B did so API documentation can be directed to him, but I am sure others might be able to chime in.

My goals for both the /docs directory/folder and the Wiki is:

1) User Documentation (How to actually use the application as a user)
2) API Documentation (how to test and use the applications' internal API)
3) Developer Documentation (Getting the project cloned/forked, installed, setting up the database, etc).

There are several issues related to documentation. The are:

#141, #150, #137

**Comment by EdwinChukwuebuka** [comment]
Okay @gbowne1

**Comment by gbowne1** [comment]
@EdwinChukwuebuka @Chizobaonorh etc 

There is also the issue of some of the other Markdown files included in this project which are not terribly accurate or complete.

- [CONTRIBUTING.md](https://github.com/gbowne1/codebooker/blob/master/CONTRIBUTING.md)
which is the documentation on how to contribute to this project including 

- [README.md](https://github.com/gbowne1/codebooker/blob/master/README.md)
which is the main project readme which should have links to relevant contribution and other documentation.

- [DEVSETUP.md](https://github.com/gbowne1/codebooker/blob/master/docs/DEVSETUP.md) This file was meant to be instructions on how to get the project installed, cloned, forked, and the database connected and how to run the app.
Right now its basically generic React instructions which are outdated

**Comment by Chizobaonorh** [comment]
> @EdwinChukwuebuka @Chizobaonorh etc
> 
> There is also the issue of some of the other Markdown files included in this project which are not terribly accurate or complete.
> 
> * [CONTRIBUTING.md](https://github.com/gbowne1/codebooker/blob/master/CONTRIBUTING.md)
>   which is the documentation on how to contribute to this project including how to contribute/do PR's, submit issues, submit discussions, clone/fork on GitHub.  I would suggest reading it too.
> * [README.md](https://github.com/gbowne1/codebooker/blob/master/README.md)
>   which is the main project readme which should have links to relevant contribution and other documentation.
> * [DEVSETUP.md](https://github.com/gbowne1/codebooker/blob/master/docs/DEVSETUP.md) This file was meant to be instructions on how to get the project installed, cloned, forked, and the database connected and how to run the app.
>   Right now its basically generic React instructions which are outdated

If its is basically generic React instructions which are outdated, what guide can i use to setup my workspace?

**Comment by Chizobaonorh** [comment]
@gbowne1 can we like have a link to your discord for easy accessibility?

**Comment by gbowne1** [comment]
I would prefer people use my Slack.  I don't use Discord for this purpose.

Slack is free.  Here is the link.

https://join.slack.com/t/connectcoders/shared_invite/zt-23ajnvegn-0nUuIlxhSDU8tQrPBL~riQ

**Comment by Believet2** [comment]
I will love to join in this project as i am looking forward to gain more experience, pls can i join the contributors? @gbowne1

**Comment by gbowne1** [comment]
@Believet2 well, we aren't really ready for documentation yet and there already is some.

**Comment by Believet2** [comment]
Ok

**Comment by bmacarini** [comment]
Hey, @gbowne1 
Is this API documentation issue still open? I didn´t find any API documentation. Is it a REST API? I would like to contributing.

**Comment by gbowne1** [comment]
@bmacarini 

- Yes its still open.  Any issues that are still left are still open.

- Theres only installation instructions 

- Yes you can work on them.

- we are putting all our documentation in the Wiki page but there is some documentation in /docs

- if you want to work on the Wiki go to the Wiki tab and clone or fork the wiki.  

- I will assign you if yiu wanted to work on this

**Comment by bmacarini** [comment]
@gbowne1 Assign this issue to me, please. I will make the API documentation.

**Comment by gbowne1** [comment]
@Ashleycodesseo I think this is pretty much complete.  I did not design the API myself so you're welcome to go through it and see if you can figure it out.

It has some issues with it

What we have now is in here. https://github.com/gbowne1/codebooker/tree/master/docs

You're welcome to work on it if you like.  Thanks

**Comment by Ashleycodesseo** [comment]
@gbowne1 It looks really good! I thought you wanted it to be added to the wiki in markdown? I can do that for you

**Comment by gbowne1** [comment]
@Ashleycodesseo yeah that needs done too if you would like to do that you're welcome to do that

**Comment by Ashleycodesseo** [comment]
@gbowne1 I'll start on it first thing in the morning. Thank you
